src/GraphQL/Introspection/__Type.cs

```diff
@@ -47,36 +48,73 @@ public __Type()
                 {
                     if (context.Source is IObjectGraphType || context.Source is IInterfaceGraphType)
                     {
-                        var includeDeprecated = context.GetArgument<bool>("includeDeprecated");
-                        var type = context.Source as IComplexGraphType;
-                        var fields = !includeDeprecated
-                            ? type.Fields.Where(f => string.IsNullOrWhiteSpace(f.DeprecationReason))
-                            : type.Fields;
-
-                        fields ??= Enumerable.Empty<FieldType>();
-                        fields = await fields.WhereAsync(f => context.Schema.Filter.AllowField(type, f)).ConfigureAwait(false);
+                        var type = (IComplexGraphType)context.Source;
+                        var fields = context.GetPooledArray<FieldType>(type.Fields.Count);
+
+                        bool includeDeprecated = context.GetArgument<bool>("includeDeprecated");
+
+                        int index = 0;
+                        foreach (var field in type.Fields.List)
```

Comment:
**A LOT** of LINQ stuff here before : chained enumerators, closures...

Reply:
Changing the introspection queries' resolvers will skew performance metrics because now the introspection query is going to show a performance improvement for _introspection_ queries, regardless of whether an equally complex query's performance has improved or not.  An introspection query executed in a live environment is relatively rare.  So now any benchmarks have less meaning.  No conclusion, just FYI.

Reply:
I know. I wanted to add "real" tests, but I haven't had time yet. If you want, you can do it yourself, then I will re-experiment. In any case, sooner or later we will need them, we have a very small set and we do not cover the whole variety of types of requests with current benchmarks.

Reply:
Introspection resolvers were a good example to test.

Reply:
Right.  It would just be nice to know what the impact is.  You can't tell now.

Reply:
Well, we can see some difference in the hero test. 👍